plugins/aws/resoto_plugin_aws/resource/apigateway.py

```diff
@@ -0,0 +1,447 @@
+        "percent_traffic": S("percentTraffic"),
+        "deployment_id": S("deploymentId"),
+        "stage_variable_overrides": S("stageVariableOverrides"),
+        "use_stage_cache": S("useStageCache"),
+    }
+    percent_traffic: int = field(default=None)
+    deployment_id: str = field(default=None)
+    stage_variable_overrides: Dict[str, str] = field(default=None)
+    use_stage_cache: bool = field(default=None)
+
+
+@define(eq=False, slots=False)
+class AwsApiGatewayStage(ApiGatewayTaggable, AwsResource):
+    kind: ClassVar[str] = "aws_api_gateway_stage"
+    mapping: ClassVar[Dict[str, Bender]] = {
+        "id": S("stageName"),
+        "name": S("stageName"),
+        "tags": S("tags", default=[]),
+        "ctime": S("createdDate"),
+        "mtime": S("lastUpdatedDate"),
+        "description": S("description"),
+        "stage_client_certificate_id": S("clientCertificateId"),
+        "stage_cache_cluster_enabled": S("cacheClusterEnabled"),
+        "stage_cache_cluster_size": S("cacheClusterSize"),
+        "stage_cache_status": S("cacheClusterStatus"),
+        "stage_method_settings": S("methodSettings"),
+        "stage_variables": S("variables"),
+        "stage_documentation_version": S("documentationVersion"),
+        "stage_access_log_settings": S("accessLogSettings"),
+        "stage_canary_settings": S("canarySettings") >> Bend(AwsApiGatewayCanarySetting.mapping),
+        "stage_tracing_enabled": S("tracingEnabled"),
+        "stage_web_acl_arn": S("webAclArn"),
+    }
+    description: Optional[str] = field(default=None)
+    stage_client_certificate_id: Optional[str] = field(default=None)
+    stage_cache_cluster_enabled: bool = field(default=None)
+    stage_cache_cluster_size: Optional[str] = field(default=None)
+    stage_cache_status: Optional[str] = field(default=None)
+    stage_method_settings: Dict[str, Dict[str, Union[bool, str, int]]] = field(default=None)
+    stage_variables: Dict[str, str] = field(default=None)
+    stage_documentation_version: Optional[str] = field(default=None)
+    stage_access_log_settings: Dict[str, str] = field(default=None)
+    stage_canary_settings: Optional[AwsApiGatewayCanarySetting] = field(default=None)
+    stage_tracing_enabled: bool = field(default=None)
+    stage_web_acl_arn: Optional[str] = field(default=None)
+    api_link: str = field(default=None)
+
+    # TODO add edge to Web Acl when applicable (via stage_web_acl_arn)
+
+    def delete_resource(self, client: AwsClient) -> bool:
+        client.call(
+            service="apigateway",
+            action="delete-stage",
+            result_name=None,
+            restApiId=self.api_link,
+            stageName=self.name,
+        )
+        return True
+
+
+@define(eq=False, slots=False)
+class AwsApiGatewayDeployment(AwsResource):
+    kind: ClassVar[str] = "aws_api_gateway_deployment"
+    reference_kinds: ClassVar[ModelReference] = {"successors": {"default": ["aws_api_gateway_stage"]}}
```

Comment:
There is no `connect_in_graph` method, but a `reference_kinds` is defined: the idea would be to have this variable everywhere you create a connection. This way it is possible to see reflect the edges created and it can be maintained/updated accodingly.

Reply:
Yeah I see. Trouble here is that the edge from Deployment to Stage is created in the RestApi model instance (line 408). I did not see how else to reflect this connection?

Reply:
Gotcha. Then I think it is fine as it is. Maybe add 2 comments: one at `reference_kinds` saying where it is defined and one in RestApi, so people do not forget to update.